src/nbl/video/CCUDADevice.cpp

```diff
@@ -1,131 +1,159 @@
 {
-	if (linksBegin+MaxAquireOps<linksEnd)
-		return CUDA_ERROR_OUT_OF_MEMORY;
-	alignas(_NBL_SIMD_ALIGNMENT) uint8_t stackScratch[MaxAquireOps*sizeof(void*)];
-
-	CUresult result = acquireResourcesFromGraphics(stackScratch,linksBegin,linksEnd,stream);
-	if (result != CUDA_SUCCESS)
-		return result;
+	auto& cu = m_handler->getCUDAFunctionTable();
+	
+	CUdeviceptr ptr = 0;
+	if (auto err = cu.pcuMemAddressReserve(&ptr, size, alignment, 0, 0); CUDA_SUCCESS != err)
+		return err;
 
-	size_t tmp = 0xdeadbeefbadc0ffeull;
-	size_t* sit = outbufferSizes;
-	for (auto iit=linksBegin; iit!=linksEnd; iit++,sit++)
+	if (auto err = cu.pcuMemMap(ptr, size, 0, memory, 0); CUDA_SUCCESS != err)
 	{
-		if (!iit->acquired)
-			return CUDA_ERROR_UNKNOWN;
+		cu.pcuMemAddressFree(ptr, size);
+		return err;
+	}
+	
+	CUmemAccessDesc accessDesc = {
+		.location = { .type = location, .id = m_handle },
+		.flags = CU_MEM_ACCESS_FLAGS_PROT_READWRITE,
+	};
 
-		result = cuda::CCUDAHandler::cuda.pcuGraphicsResourceGetMappedPointer_v2(&iit->asBuffer.pointer,outbufferSizes ? sit:&tmp,iit->cudaHandle);
-		if (result != CUDA_SUCCESS)
-			return result;
+	if (auto err = cu.pcuMemSetAccess(ptr, size, &accessDesc, 1); CUDA_SUCCESS != err)
+	{
+		cu.pcuMemUnmap(ptr, size);
+		cu.pcuMemAddressFree(ptr, size);
+		return err;
 	}
+
+	*outPtr = ptr;
+
 	return CUDA_SUCCESS;
 }
-CUresult CCUDAHandler::acquireAndGetMipmappedArray(GraphicsAPIObjLink<video::IGPUImage>* linksBegin, GraphicsAPIObjLink<video::IGPUImage>* linksEnd, CUstream stream)
+
+CUresult CCUDADevice::createSharedMemory(
+	core::smart_refctd_ptr<CCUDASharedMemory>* outMem, 
+	CCUDASharedMemory::SCreationParams&& inParams)
 {
-	if (linksBegin+MaxAquireOps<linksEnd)
-		return CUDA_ERROR_OUT_OF_MEMORY;
-	alignas(_NBL_SIMD_ALIGNMENT) uint8_t stackScratch[MaxAquireOps*sizeof(void*)];
+	if (!outMem)
+		return CUDA_ERROR_INVALID_VALUE;
 
-	CUresult result = acquireResourcesFromGraphics(stackScratch,linksBegin,linksEnd,stream);
-	if (result != CUDA_SUCCESS)
-		return result;
+	CCUDASharedMemory::SCachedCreationParams params = { inParams };
 
-	for (auto iit=linksBegin; iit!=linksEnd; iit++)
-	{
-		if (!iit->acquired)
-			return CUDA_ERROR_UNKNOWN;
+	auto& cu = m_handler->getCUDAFunctionTable();
 
-		result = cuda::CCUDAHandler::cuda.pcuGraphicsResourceGetMappedMipmappedArray(&iit->asImage.mipmappedArray,iit->cudaHandle);
-		if (result != CUDA_SUCCESS)
-			return result;
-	}
-	return CUDA_SUCCESS;
-}
-CUresult CCUDAHandler::acquireAndGetArray(GraphicsAPIObjLink<video::IGPUImage>* linksBegin, GraphicsAPIObjLink<video::IGPUImage>* linksEnd, uint32_t* arrayIndices, uint32_t* mipLevels, CUstream stream)
-{
-	if (linksBegin+MaxAquireOps<linksEnd)
-		return CUDA_ERROR_OUT_OF_MEMORY;
-	alignas(_NBL_SIMD_ALIGNMENT) uint8_t stackScratch[MaxAquireOps*sizeof(void*)];
+	uint32_t metaData[16] = { 48 };
```

Comment:
mind moving this holy constant to a single variable (can be static and global on top of this file)